### ISSUE TITLE
Maintain left-to-right order

### ISSUE THREAD

**Issue by desandro** [desandro/masonry#873]
Lots of people have requested this over time. I'd like to see how many are interested. **Add a 👍  reaction to this issue** if you would like to see this feature added to Masonry and Isotope. Subscribe to this issue using the button in the sidebar.

The Masonry layout algorithm works by looking at vertical space. However you may want keep the left-to-right order of the items.

![masonry-ltr](https://cloud.githubusercontent.com/assets/85566/16620055/f812f61a-435e-11e6-9265-2d115b0caf58.png)

In the past I've recommended [Masonry Ordered](https://masonry-ordered.tasuki.org/), but it's now out of date and no longer compatible with current version of Masonry.

**Comment by timbouchard** [comment]
👍

**Comment by egucciar** [comment]
I was just looking for this! without this, may have to deal with a plain-old regular grid layout. But we want animations! :)

**Comment by markmaurerrr** [comment]
+1

**Comment by egucciar** [comment]
any ideas on how to implement. desperately need this feature as if the heights are not pixel-perfect our items keep swapping order and its disorienting. need this feature.

**Comment by clinto69** [comment]
I'd pay you right now to have this ferature in. Im desperate!

**Comment by Shakhmurat** [comment]
oh yeah , it is very, very necessary option! I can not solve this problem for a week.

**Comment by egucciar** [comment]
If anyone needs a temporary quick fix , make sure all items you want in the same row have the same height and it will layout in the right order .

**Comment by miguelxt** [comment]
👍

**Comment by cavoirom** [comment]
This is a dirty patch can temporary solve the issue:
1. remove these lines of code:
`var minimumY = Math.min.apply( Math, colGroup );
var shortColIndex = colGroup.indexOf( minimumY );`

2. Add these line of code at the same position:
`var shortColIndex = this.items.indexOf(item) % this.cols;
var minimumY = colGroup[shortColIndex];`

The idea is place the item by the order in its list and don't care the height, you need to make sure all items have the same with.

**Comment by jokke150** [comment]
This would be really great. 

As there seems to be a certain demand for this feature, will this feature request be implemented in near future? :)

**Comment by desandro** [comment]
Thanks for all your input. I have been working on this feature. I have no current ETA. I'll follow up once a release is ready.

**Comment by desandro** [comment]
[Masonry v4.2.0 has added `horizontalOrder`](http://masonry.desandro.com/options.html#horizontalorder). You can now maintain left-to-right with Masonry layouts.

Thanks all for your feedback over the years (!). Glad to have this feature finally delivered. Please open a new issue if you run into trouble. 🌈🐻

**Comment by jimmyadaro** [comment]
Is this available for Isotope as well?

**Comment by desandro** [comment]
Yes, [horizontalOrder has been added as an option for the masonry layout mode for Isotope](http://isotope.metafizzy.co/layout-modes/masonry.html#horizontalorder)

**Comment by jimmyadaro** [comment]
Amazing! Thank you very much.

**Comment by derek-knox** [comment]
Right on! Nice work as always.

**Comment by moequraishi** [comment]
Hi,

It seems like the horizontalOrder does not work properly. It seems to break the layout of the columns if one of the columns is bigger than the rest....

**Comment by javarch** [comment]
Ich werde ab  09.03.2018 nicht im Büro sein. Ich kehre zurück am
14.03.2018.
Ich werde Ihre Nachricht nach meiner Rückkehr beantworten.

Die Rückmeldung bezieht sich auf ein Mail mit folgendem Thema:
Re: [desandro/masonry] Maintain left-to-right order (#873)

____________________________________________________________________________________________
Gesendet (c) GRZ/RACON Linz 2018 Agent 'Abwesenheit'

----------------------------------------
Raiffeisen Software GmbH
Firmenbuchnummer: 86804d, Landesgericht Linz
DVR: 0572934, UID: ATU 36961105

Der Austausch von Nachrichten mit dem oben angeführten Absender via E-Mail
dient ausschließlich Informationszwecken. Rechtsgeschäftliche Erklärungen
dürfen über dieses Medium nicht ausgetauscht werden.
Correspondence with above mentioned sender via e-mail is only for
information purposes. This medium is not to be used for the exchange of
legally-binding communications.
----------------------------------------

**Comment by markmaurerrr** [comment]
@javarch hahaha, nice one!
